### ISSUE TITLE
[BUG] Chrome device not listed

### ISSUE THREAD

**Issue by gruvw** [nvim-flutter/flutter-tools.nvim#306]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Chrome device is not listed when using `:FlutterDevices` but is when using `flutter devices`.

### Expected Behavior

Chrome decive should be listed in neovim too.

### Steps To Reproduce

N/A

### Environment

```Markdown
- OS: Ubuntu 22.04 LTS
- Flutter version: 3.13.8
- Is flutter in $PATH: yes
- neovim version: v0.9.2
```

**Comment by gruvw** [comment]
Still no solution to this day...

Using flutter-tools:
![image](https://github.com/akinsho/flutter-tools.nvim/assets/63407038/44a34664-6b3a-468a-b351-8b6b6fb342ee)

Running `flutter devices` in terminal:
![image](https://github.com/akinsho/flutter-tools.nvim/assets/63407038/e58f96d3-63ac-49eb-a0ca-475035d0abdf)

**Comment by sidlatau** [comment]
I am unable to reproduce this issue. The code appears to call the same flutter devices command and then parses the results. Here is the code for reference:
```lua
function M.parse(line, device_type)
  if line:find("Manufacturer") and line:find("Platform") then return end
  if line:find("crashdata") then return end
  local parts = vim.split(line, "•")
  local is_emulator = device_type == EMULATOR
  local name_index = not is_emulator and 1 or 2
  local id_index = not is_emulator and 2 or 1
  if #parts == 4 then
    return {
      name = vim.trim(parts[name_index]),
      id = vim.trim(parts[id_index]),
      platform = vim.trim(parts[3]),
      system = vim.trim(parts[4]),
      type = device_type,
    }
  end
end
```
@gruvw, are you still experiencing this issue? If so, could you provide the full output of the `flutter devices` command? It's possible that the output contains more than three "•" symbols in some lines, which might be causing the parsing problem.

**Comment by gruvw** [comment]
Hello, thank you @sidlatau for taking a look. Yes, I am still experiencing this issue.

Here is the full output of running the cmd `flutter devices` on my terminal:

```bash
flutter devices
Found 2 connected devices:
  Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.5 LTS 6.8.0-109048-tuxedo
  Chrome (web)    • chrome • web-javascript • Brave Browser 131.1.73.101

Run "flutter emulators" to list and start any available device emulators.

If you expected another device to be detected, please run "flutter doctor" to diagnose potential issues. You may also try increasing the time to wait for
connected devices with the "--device-timeout" flag. Visit https://flutter.dev/setup/ for troubleshooting tips.
```

Meanwhile, on the Neovim interface I only see the `Linux (desktop)` device for launch when executing `:FlutterDevices`:

![image](https://github.com/user-attachments/assets/8880955e-7b4e-464a-a89c-245b03e10db5)

I am seeing this issue on the following version: `flutter-tools.nvim - url https://github.com/nvim-flutter/flutter-tools.nvim - branch main - commit 54314bc`.

If you need anything else please let me know, I would be happy to help more as this issue is quite a problem for me.

**Comment by gruvw** [comment]
After a little bit more investigation, it does not seem that it is a parsing error from the code you referenced in https://github.com/nvim-flutter/flutter-tools.nvim/issues/306#issuecomment-2563873469.

I modified the code of `get_devices` to show the results in a vim notification:

```lua
---@param result string[]
---@param type integer
local function get_devices(result, type)
  local lines = ""
  local devices = {}
  for _, line in pairs(result) do
    lines = lines .. line .. "\n"
    local device = M.parse(line, type)
    if device then
      table.insert(devices, device)
      if type == EMULATOR and device.system and device.system == "android" then
        local cold_boot_device = vim.tbl_extend("force", {}, device, { cold_boot = true })
        cold_boot_device.name = fmt("%s (cold boot)", device.name)
        table.insert(devices, cold_boot_device)
      end
    end
  end
  vim.notify(lines, vim.log.levels.ERROR)
  return devices
end
```

And in Neovim, I indeed only see one device/line instead of the two I get when executing the command from the terminal:

![image](https://github.com/user-attachments/assets/aa50db8b-abcc-4837-9f7c-1b68c64ff20b)
 
I am not sure of why that could be as in VSCode for example all the devices are correctly recognized, and I can also (of course) run the Flutter project in the WEB using the terminal directly.

**Comment by gruvw** [comment]
Even the Neovim embedded terminal shows me two devices:

![image](https://github.com/user-attachments/assets/c74e5595-e705-4c0e-a690-0b215e9c44e0)

**Comment by sidlatau** [comment]
Hmm, @gruvw  could you print `cmd` argument here: https://github.com/nvim-flutter/flutter-tools.nvim/blob/54314bcb6856dfd31a500226587c95402122e29f/lua/flutter-tools/devices.lua#L157 ?

The output should be the same as when running it directly. It’s strange that the results are different.

**Comment by gruvw** [comment]
The print of the cmd argument gives `/home/gruvw/snap/flutter/common/flutter/bin/flutter`.

I ran the full `/home/gruvw/snap/flutter/common/flutter/bin/flutter devices` to double-check, but I indeed still have the two devices when ran in terminal, but not in Neovim.

**Comment by gruvw** [comment]
As comparison, a python shell gives the correct output when invoking the same cmd:

```
python3
Python 3.10.12 (main, Nov  6 2024, 20:22:13) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> 
>>> subprocess.run(["/home/gruvw/snap/flutter/common/flutter/bin/flutter", "devices"])
Found 2 connected devices:
  Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.5 LTS
  6.8.0-109048-tuxedo
  Chrome (web)    • chrome • web-javascript • Brave Browser 131.1.73.101

Run "flutter emulators" to list and start any available device emulators.

If you expected another device to be detected, please run "flutter doctor" to
diagnose potential issues. You may also try increasing the time to wait for
connected devices with the "--device-timeout" flag. Visit
https://flutter.dev/setup/ for troubleshooting tips.
CompletedProcess(args=['/home/gruvw/snap/flutter/common/flutter/bin/flutter', 'devices'], returncode=0)
>>> 
```

**Comment by gruvw** [comment]
Ok, I have isolated the problem after a some more poking around and using the Neovim command `:lua vim.notify(io.popen("flutter devices"):read("*a"))`.

I found out that it was only showing the incomplete list when I was opening Neovim by using my Qtile cmd shortcut (using `lazy.spawn` https://docs.qtile.org/en/v0.17.0/manual/config/lazy.html), which is basically the way I open Neovim all the time. However, the list of devices is complete when I open Neovim in the terminal (by executing `nvim`).

I still have no idea why it makes a difference, but at least I know this has nothing to do with `nvim-flutter` as when opening `nvim` from my terminal the devices list is complete.
